airflow/utils/email.py

```diff
@@ -226,9 +228,9 @@ def send_mime_email(
             if smtp_starttls:
                 conn.starttls()
             if smtp_user and smtp_password:
-                conn.login(smtp_user, smtp_password)
+                conn.login(smtp_user if smtp_user else "", smtp_password)
             log.info("Sent an alert email to %s", e_to)
-            conn.sendmail(e_from, e_to, mime_msg.as_string())
+            conn.sendmail(e_from if e_from else "", e_to, mime_msg.as_string())
```

Comment:
I don't think this is actually valid -- SMPT requires an address in  "MAIL FROM".

Reply:
Yep. Looked at the code of the library and it would fail. Throwing Runtime Errors instead.